assets/js/custom/UserNotifications.js

```diff
@@ -26,35 +40,182 @@ class UserNotifications {
+        self.reactions = true
+        if (self.reactionNotificationsLoaded < self.notificationsFetchCount) {
+          self.fetchMoreNotifications(self.notificationsFetchCount, self.reactionNotificationsLoaded,
+            'reaction', 'reaction-notification-', $('#reaction-notifications'))
+        }
       }
     })
     $(document).on('click', '#remix-notif', function () {
       if (self.remixes === false) {
         self.resetChips()
         self.selectChip('remix-notif', self.remixes)
+        self.remixes = true
+        if (self.remixNotificationsLoaded < self.notificationsFetchCount) {
+          self.fetchMoreNotifications(self.notificationsFetchCount, self.remixNotificationsLoaded,
+            'remix', 'remix-notification-', $('#remix-notifications'))
+        }
       }
     })
+
+    $(document).on('scroll', function () {
+      const position = $(window).scrollTop()
+      const bottom = $(document).height() - $(window).height()
+      const pctVertical = position / bottom
+      if (pctVertical >= 0.7) {
+        if (self.all === true) {
+          self.fetchMoreNotifications(self.notificationsFetchCount, self.allNotificationsLoaded,
+            'all', 'catro-notification-', $('#notifications'))
+        } else if (self.follower === true) {
+          self.fetchMoreNotifications(self.notificationsFetchCount, self.followerNotificationsLoaded,
+            'follow', 'follow-notification-', $('#follow-notifications'))
+        } else if (self.comment === true) {
+          self.fetchMoreNotifications(self.notificationsFetchCount, self.commentNotificationsLoaded,
+            'comment', 'comment-notification-', $('#comment-notifications'))
+        } else if (self.reactions === true) {
+          self.fetchMoreNotifications(self.notificationsFetchCount, self.reactionNotificationsLoaded,
+            'reaction', 'reaction-notification-', $('#reaction-notifications'))
+        } else if (self.remixes === true) {
+          self.fetchMoreNotifications(self.notificationsFetchCount, self.remixNotificationsLoaded,
+            'remix', 'remix-notification-', $('#remix-notifications'))
+        }
+      }
+    })
+  }
+
+  fetchMoreNotifications (limit, loadedCount, type, idPrefix, $container) {
+    const self = this
+    if (this.empty === true || this.fetchActive === true) {
+      return
+    }
+    this.fetchActive = true
+    $.ajax({
+      url: self.fetchNotificationsUrl + '/' + limit + '/' + loadedCount + '/' + type,
+      type: 'get',
+      success: function (data) {
+        data['fetched-notifications'].forEach(function (fetched) {
+          self.generateNotificationBody(fetched, idPrefix, $container)
+        })
+
+        self.updateLoadedNotificationsCount(type, data['fetched-notifications'].length)
+        self.fetchActive = false
+      },
+      error: function (xhr) {
+        self.handleError(xhr)
+      }
+    })
+  }
+
+  updateLoadedNotificationsCount (type, fetchedAmount) {
```

Comment:
instead of counting all values at all time (even in the beginning)
you could just check the notification containers for its direct children to get the count before fetching

E.g. #comment-notifications